Sources/Controllers/Map/Toolbars/OAWeatherToolbar.mm

```diff
@@ -419,18 +461,200 @@ - (void)updateWidgetsInfo
     }
 }
 
+#pragma mark - Start/stop button
+
+- (IBAction)onPlayForecastClicked:(id)sender
+{
+    EOAWeatherToolbarAnimationState animationState = _animationState == EOAWeatherToolbarAnimationStateIdle ? EOAWeatherToolbarAnimationStateStarted : EOAWeatherToolbarAnimationStateIdle;
+    _animationState = animationState;
+    if (animationState == EOAWeatherToolbarAnimationStateStarted)
+    {
+        _currentStep = [self.timeSliderView getIndexForOptionStepsAmountWithoutDrawMark];
+        _animationStartStep = _currentStep;
+        _selectedDate = _timeValues[_currentStep];
+        
+        _animateStepCount = kForecastMaxStepsCount;
+        NSInteger remainingStepsForMidnight = _timeValues.count - _currentStep - 1;
+        if (_animateStepCount > remainingStepsForMidnight)
+            _animateStepCount = remainingStepsForMidnight;
+        _animationStartStepCount = _animateStepCount;
+        
+        [_plugin prepareForDayAnimation:_selectedDate];
+        [self updateSelectedDate:_selectedDate forAnimation:YES resetPeriod:YES];
+        [self scheduleAnimationStart];
+    }
+    else
+    {
+        [self stopAnimation];
+    }
+    [self updatePlayForecastButton];
+}
+
+- (void) updatePlayForecastButton
+{
+    NSString *iconName = _animationState == EOAWeatherToolbarAnimationStateIdle ? @"ic_custom_play" : @"ic_custom_pause";
+    [_playButton setImage:[UIImage templateImageNamed:iconName] forState:UIControlStateNormal];
+}
+
+- (void) scheduleAnimationStart
+{
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        
+        [NSThread sleepForTimeInterval:kAnimationStartDelaySec];
```

Comment:
Blocking the system queue is bad practice. At the very least, you might consider using the following approach:
```
- (void)scheduleAnimationStart {
    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(kAnimationStartDelaySec * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
            [self startAnimationLoop];
        });
    });
}
```

Reply:
https://github.com/osmandapp/OsmAnd-iOS/pull/3884/commits/f124ab0b19d5259133d65049264c52c081341c2a

Reply:
It is better to use CADisplayLink here (for animation). Please look how it work in MGSwipeTableCell.